lib/config.js

```diff
@@ -211,11 +210,22 @@ function normalizeArrays (opts) {
 
 function normalizeBools (opts) {
   BOOL_OPTS.forEach(function (key) {
-    if (key in opts) opts[key] = normalizeBool(opts[key])
+    if (key in opts) opts[key] = strictBool(opts[key])
   })
 }
 
 function truncateOptions (opts) {
   if (opts.serviceVersion) opts.serviceVersion = truncate(String(opts.serviceVersion), config.INTAKE_STRING_MAX_SIZE)
   if (opts.hostname) opts.hostname = truncate(String(opts.hostname), config.INTAKE_STRING_MAX_SIZE)
 }
+
+function strictBool (value) {
+  if (typeof value === 'boolean') {
+    return value
+  }
+  // This will return undefined for unknown inputs, resulting in them being skipped.
```

Comment:
We should probably log a warning that the value was invalid and that we fell back to the default

Reply:
fixed.